packages/components/pie-webc-core/src/types/component-default-props-generic.ts

```diff
@@ -1 +1,24 @@
-export type ComponentDefaultPropsGeneric<T, K extends keyof T> = Readonly<Required<Pick<T, K>>>;
+/**
+ * This type should be used when defining the default props for a component.
+ * It is a generic type that takes two type parameters:
+ * - `T` is the type of the props object that the default props are being defined for.
+ * - `K` is the type of the keys in `T` that should be required in the default props.
+ * By default, `K` is set to be all the keys in `T`. This means that all the keys in `T` will be required in the default props.
+ * You can override this by specifying a subset of the keys in `T` that should be required in the default props.
+ *
+ * @example ```tsx
+ * interface MyComponentProps {
+ *    a: string;
+ *   b?: number;
+ * }
+ * const allProps: ComponentDefaultProps<MyComponentProps> = {
+ *   a: 'default value',
+ *   b: 42,
+ * };
+ *
+ * const someProps: ComponentDefaultProps<MyComponentProps, 'a'> = {
```

Comment:
Might be worth adding the omit version example, I know that's separate to this method as it's using a utility but could be worth explaining / showing.

Reply:
Good shout, updated